eZ/Publish/API/Repository/ContentService.php

```diff
@@ -439,6 +439,28 @@ public function deleteTranslation(ContentInfo $contentInfo, $languageCode);
      */
     public function deleteTranslationFromDraft(VersionInfo $versionInfo, $languageCode);
 
+    /**
+     * Hides Content by making all the Locations appear hidden.
+     * It does not persist hidden state on Location object itself.
+     *
+     * Content hidden by this API can be revealed by revealContent API.
+     *
+     * @see revealContent
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     */
+    public function hideContent(ContentInfo $contentInfo): void;
+
+    /**
+     * Reveals Content hidden by hideContent API.
+     * Locations which were hidden before hiding Content will remain hidden.
```

Comment:
i don't understand this... say you have a content with two locations. you hide that content => two locations hidden. 
Shouldn't revealContent will make those locations not hidden? what that 'begore hidden Content` means? do we have a timestamp for that?

Reply:
@SylvainGuittard could you explain the business logic behind restoring locations to original state after content reveal?

Reply:
> Shouldn't revealContent will make those locations not hidden? what that 'before hidden Content` means? do we have a timestamp for that?

Hmm, it actually means exactly what it says. If you've hidden just one of the Locations prior calling this, after calling `hideContent`, both Locations will appear hidden on API level. After calling `revealContent`, you'll get "previous state" - one of the Locations will be hidden. It's just about `isHidden` on `ContentInfo` level taking priority over `isHidden` on `Location` level.

Keeping state of hidden/visible Locations prior calling this API is a key part of this feature requested by Editors.

Reply:
@crevillo 

> what that 'before hidden Content` means?

One of the concern that community members had when we talked about this feature (https://discuss.ezplatform.com/t/we-want-your-feedback-on-unpublish/524) was that if we change the visibility of all locations, it will be impossible to restore the initial status
(https://discuss.ezplatform.com/t/we-want-your-feedback-on-unpublish/524/3).

During the spike, our engineering team found a way to store this information. 
As @lserwatka said below:
>during reveal process we will restore original state on all locations (some might be hidden, some not).

Reply:
I see this a bit tricky... 
Let's say i have a content with two locations, both of them *not* hidden
1) When i hide content, will both locations be *hidden* or the locations objects will still have is_hidden to `true`?
1a) if we don't hide the locations, where we're adding the logic to return a 404 in case content is hidden?
1b) if locations are hidden, when i reveal the content, do i need to unhide at least one location to make the content visible somewhere?

Reply:
@crevillo Content visibility has higher priority over Location visibility. If you hide content, the visibility of location nodes becomes irrelevant, yet the original visibility values remain unchanged. It is done on a domain model ([DomainMapper::mapLocation](https://github.com/ezsystems/ezpublish-kernel/pull/2549/files#diff-4043c8073a9a8c2593b809f674fafa42R524)) level.

To answer your questions:

1) Yes, the Locations will be hidden on a domain model level (check the Domain Mapper).
1a) They'll be hidden.
1b) You don't. Original visibility of locations remain unchanged. Unless you had all of the locations hidden prior to hiding content (which made content not visible anywhere), you don't have to change location visibility.